spec/controllers/invoices/pdf_controller_spec.rb

```diff
@@ -9,44 +9,34 @@
     sign_in employee
   end
 
-  describe 'GET #new' do
+  describe 'GET #show' do
     let(:pdf_double) { instance_double Invoices::PDF }
+    let(:pdf_data) { double('data') }
 
     before do
       allow(Invoices::PDF).to receive(:new).and_return pdf_double
+      allow(pdf_double).to receive(:read).and_return pdf_data
       allow(pdf_double).to receive(:persisted?).and_return true
       allow(pdf_double).to receive(:persisted_pdf_path).and_return 'persisted.pdf'
     end
 
     it 'returns pdf path' do
-      expect(controller).to receive(:send_file).with('persisted.pdf', filename: 'Rechnung Hans 2017-09-04.pdf')
+      expect(controller).to receive(:send_data).with(pdf_data, filename: 'Rechnung Hans 2017-09-04.pdf')
 
-      get :new, format: :pdf, params: { invoice_id: invoice.to_param }
-    end
-
-    context 'with not persisted pdf' do
-      before do
-        expect(pdf_double).to receive(:persisted?).and_return false
-        expect(pdf_double).to receive(:create_temporary).and_return 'temporary.pdf'
-      end
-
-      it 'returns temporary pdf path' do
-        expect(controller).to receive(:send_file).with('temporary.pdf', filename: 'Rechnung Hans 2017-09-04.pdf')
-
-        get :new, format: :pdf, params: { invoice_id: invoice.to_param }
-      end
+      get :show, format: :pdf, params: { invoice_id: invoice.to_param }
     end
 
     context 'with templated = true' do
       let(:templated_pdf_double) { instance_double Invoices::TemplatedPDF }
+      let(:templated_pdf_data) { double('templated data') }
 
       it 'returns templated pdf path' do
-        expect(Invoices::TemplatedPDF).to receive(:new).with('persisted.pdf').and_return templated_pdf_double
-        expect(templated_pdf_double).to receive(:path).and_return 'templated.pdf'
+        expect(Invoices::TemplatedPDF).to receive(:new).with(pdf_data).and_return templated_pdf_double
```

Comment:
RSpec/MessageSpies: Prefer `have_received` for setting message expectations. Setup `Invoices::TemplatedPDF` as a spy using `allow` or `instance_spy`.